lib/std/tar.zig

```diff
@@ -1,172 +1,1134 @@
+                    hdr.atime = unixTime(try parseNumeric(star.accessTime()), 0);
+                    hdr.ctime = unixTime(try parseNumeric(star.changeTime()), 0);
+                } else if (format.contains(.gnu)) {
+                    hdr.fmt = format;
+                    const gnu = v7.gnu();
+                    {
+                        const b = gnu.accessTime();
+                        if (b[0] != 0)
+                            hdr.atime = unixTime(try parseNumeric(b), 0);
+                    }
+                    {
+                        const b = gnu.changeTime();
+                        if (b[0] != 0)
+                            hdr.ctime = unixTime(try parseNumeric(b), 0);
+                    }
                 }
-            },
-            .normal => {
-                if (file_size == 0 and unstripped_file_name.len == 0) return;
-                const file_name = try stripComponents(unstripped_file_name, options.strip_components);
+                if (prefix.len > 0) {
+                    var fbs = std.io.fixedBufferStream(self.name_buf);
+                    const w = fbs.writer();
+                    _ = try w.write(prefix);
+                    try w.writeByte(fs.path.sep);
+                    _ = try w.write(hdr.name);
+                    self.name_buf[prefix.len + hdr.name.len + 1] = 0;
+                }
+            }
+            return hdr;
+        }
+    };
+}
+
+const Pax = struct {
+    const path = "path";
+    const linkpath = "linkpath";
+    const size = "size";
+    const uid = "uid";
+    const gid = "gid";
+    const uname = "uname";
+    const gname = "gname";
+    const mtime = "mtime";
+    const atime = "atime";
+    const ctime = "ctime";
+
+    const schily_xattr = "SCHILY.xattr.";
+
+    // Keywords for GNU sparse files in a PAX extended header.
+    const Gnu = struct {
+        const sparse = "GNU.sparse.";
+        const sparse_num_blocks = "GNU.sparse.numblocks";
+        const sparse_offset = "GNU.sparse.offset";
+        const sparse_num_bytes = "GNU.sparse.numbytes";
+        const sparse_map = "GNU.sparse.map";
+        const sparse_name = "GNU.sparse.name";
+        const sparse_major = "GNU.sparse.major";
+        const sparse_minor = "GNU.sparse.minor";
+        const sparse_size = "GNU.sparse.size";
+        const sparse_real_size = "GNU.sparse.realsize";
+    };
+};
+
+/// Copies `file_path` to `buf` and adds a null terminator
+pub fn toPosixPath(file_path: []const u8, comptime buf_len: u32, buf: *[buf_len]u8) ![:0]u8 {
+    if (std.debug.runtime_safety) assert(std.mem.indexOfScalar(u8, file_path, 0) == null);
+    // >= rather than > to make room for the null byte
+    if (file_path.len >= buf_len) return error.NameTooLong;
+    mem.copy(u8, buf, file_path);
+    buf[file_path.len] = 0;
+    return buf[0..file_path.len :0];
+}
+
+/// return the most significant, 'top' half of the time as an i64
+fn truncateTime(t: i128) i64 {
+    return @bitCast([2]i64, t)[0];
```

Comment:
I believe this is (unnecessarily) endian-specific; IMO the more idiomatic formulation is `@intCast(i64, t >> 64)`, which will work everywhere. (Shifting 1-bits out to the right using `>>` is currently allowed.)

Reply:
Thanks! Fixed.